plugin.video.worldstarhiphop/resources/lib/worldstarhiphop_play.py

```diff
@@ -0,0 +1,160 @@
+            soup = getSoup(html_source)
+
+            # log("html_source", html_source)
+
+            # A bit of a dirty hack, but let's try it anyway...
+            # so.addVariable("file","http://hw-videos.worldstarhiphop.com/u/vid/2015/09/SAWGSqGpaohk.mp4");
+            # or
+            # <source src="http://hw-videos.worldstarhiphop.com/u/vid/2017/04/Gtlg3yKHNNqP.mp4" type="video/mp4">
+            pos_vid_url = str(html_source).find("hw-videos.worldstarhiphop.com/")
+            if pos_vid_url >= 0:
+                pos_start_quote = str(html_source).rfind('"', 0, pos_vid_url)
+                pos_end_quote = str(html_source).find('"', pos_start_quote + 1)
+                video_url = html_source[pos_start_quote + 1: pos_end_quote]
+                have_valid_url = True
+            else:
+                # Maybe it's a youtube video then ?!
+                if str(html_source).find("www.youtube.com/embed") >= 0:
+                    # Seems like it's an youtube video
+                    # <iframe src="http://www.youtube.com/embed/1xcxn7pOYCg?autoplay=1" width="640" height="390" frameborder="0"></iframe>
+                    # look for http youtube
+                    video_urls = soup.findAll('iframe', attrs={'src': re.compile("^http://www.youtube.com/embed")}, limit=1)
+                    if len(video_urls) == 0:
+                        # look for https youtube
+                        video_urls = soup.findAll('iframe', attrs={'src': re.compile("^https://www.youtube.com/embed")}, limit=1)
+                        if len(video_urls) == 0:
+                            unplayable_media_file = True
+                        else:
+                            video_url = video_urls[0]['src']
+
+                            log("video_url", video_url)
+
+                            # make youtube plugin url
+                            pos_of_last_question_mark = video_url.rfind("?")
+                            video_url = video_url[0: pos_of_last_question_mark]
+                            video_url_len = len(video_url)
+                            youtubeID = video_url[len("https://www.youtube.com/embed/"):video_url_len]
+                            youtube_url = 'plugin://plugin.video.youtube/play/?video_id=%s' % youtubeID
+                            have_valid_url = True
+                            video_url = youtube_url
+                    else:
+                        video_url = video_urls[0]['src']
+
+                        log("video_url", video_url)
+
+                        # make youtube plugin url
+                        pos_of_last_question_mark = video_url.rfind("?")
+                        video_url = video_url[0: pos_of_last_question_mark]
+                        video_url_len = len(video_url)
+                        youtubeID = video_url[len("http://www.youtube.com/embed/"):video_url_len]
+                        youtube_url = 'plugin://plugin.video.youtube/play/?video_id=%s' % youtubeID
```

Comment:
you might want to support tubed too

Reply:
What do u mean with this?

Reply:
Tubed is a new YouTube add-on available for matrix: https://forum.kodi.tv/showthread.php?tid=358806

Reply:
I'll look into it in the future.